app/controllers/templates_controller.rb

```diff
@@ -92,7 +92,9 @@ def preview
       return
     end
     @template.template = params[:template]
-    safe_render(@template, Foreman::Renderer::PREVIEW_MODE, escape_json: true)
+
+    renderer = params.delete('safemode') ? Foreman::Renderer::SafeModeRenderer : Foreman::Renderer
+    safe_render(@template, Foreman::Renderer::PREVIEW_MODE, renderer, escape_json: true)
```

Comment:
I mean imagine scenario where a regular user is able to craft a template with same code like `Kernel.spawn` whatever. And then even when safemode is enforced on global setting he/she can execute it (even through preview). This is security issue.

Do we want only to allow enabling safemode (not disabling)? (Maybe this is what the PR does lemme check).

Reply:
If I get this correctly, `SafeMode` is safe to use in any case.
So this either uses `SafeMode` if the parameter is set or (UnsafeMode if it's active | SafeMode if it's active).
So I believe you can only render in unsafe mode if that's enabled via a global setting.

Reply:
I believe there is no way to force the unsafe mode. If the `safemode` parameter is passed, the template is rendered using `Foreman::Renderer::SafeModeRenderer`, otherwise using `Foreman::Renderer` (which depends on the setting).

Reply:
Yes I agree we are good.